### PR TITLE
boards: frdm_mcxc444: Add counters support

### PR DESCRIPTION
**Pull request by NeilChen93**
mcxc444 also has pit and rtc counters, add the counters to board documentation, and enable it explicitly in board dts. Set rtc clock to 32 kHz oscillator.

### CODE REVIEW

**Review by danieldegrasse — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@NeilChen93, please fix the merge conflict. Also kindly provide details on how these counters were tested.

**Review by dleach02 — DISMISSED** [review]

File: doc/releases/release-notes-4.1.rst [hunk 1]
```diff
@@ -93,6 +93,8 @@ Boards & SoC Support
 
 * Made these board changes:
 
+  * Enabled RTC, PIT on NXP ``frdm_mcxc444``
```
Comment (mmahadevan108) [inline doc/releases/release-notes-4.1.rst:96]: @NeilChen93 , apologies for the back and forth on this. Upon further discussion with the 4.1 release managers it was decided that we do not want to include such changes in the release notes. Kindly delete this addition.

Comment (dleach02) [reply doc/releases/release-notes-4.1.rst]: I dropped the commit for you @NeilChen93

**Review by dleach02 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

File: doc/releases/release-notes-4.1.rst (hunk 1 above)
Comment (NeilChen93) [reply doc/releases/release-notes-4.1.rst]: Thanks

**Pull request merged**